pysatMadrigal/instruments/methods/madrigal.py

```diff
@@ -491,14 +490,13 @@ def good_exp(exp, date_array=None):
         if date_array is None:
             gflag = True
         else:
-            exp_start = dt.datetime(exp.startyear, exp.startmonth,
-                                    exp.startday, exp.starthour,
-                                    exp.startmin, exp.startsec)
-            exp_end = dt.datetime(exp.endyear, exp.endmonth, exp.endday,
-                                  exp.endhour, exp.endmin, exp.endsec)
+            exp_start = dt.date(exp.startyear, exp.startmonth,
+                                exp.startday)
+            exp_end = (dt.date(exp.endyear, exp.endmonth, exp.endday)
+                       + dt.timedelta(days=1))
 
             for date_val in date_array:
-                if date_val >= exp_start and date_val < exp_end:
+                if date_val.date() >= exp_start and date_val.date() <= exp_end:
```

Comment:
This was one of the sources of the bug.  If the experiment file starts partway through the first requested day, using datetime to compare will fail. ie, the code was being asked to evaluate `date_val >= exp_start` as
```
dt.datetime(2010, 4, 12, 0, 0, 0) >= dt.datetime(2010, 4, 12, 18, 34, 27)
```
Converting both to date and adding a pad for the end point to solve this issue.